src/scripting/game_lua_kernel.cpp

```diff
@@ -3248,16 +3246,135 @@ int game_lua_kernel::intf_log_replay(lua_State* L)
 	return 0;
 }
 
-/** Adding new events */
+struct lua_event_filter : public game_events::event_filter
```

Comment:
The `lua_event_filter` and `filter_lua` classes seem to be almost the same, however they take different approaches to accessing the lua_kernel. This one stores a reference, the other one accesses it via `resources::`.

I'm not saying which of those is better, but I'm interested in why they're inconsistent.

Reply:
They basically are the same… I guess the only reason it exists twice is because they are file-local classes.

One is used for filter code placed in an `[event]` tag, while the other is used when you add an event from Lua and pass a function as the filter.

Reply:
This links into the comment that I clicked "resolve" on instead of posting - storing a reference vs making the `resources::lua_kernel` non-nullptr are two different ways to access the kernel during play_controller's destruction sequence; both of them seem to be changing from the existing design.

Reply:
Resolving because the `filter_lua` class no longer exists.